src/coreclr/jit/codegencommon.cpp

```diff
@@ -8177,6 +8201,15 @@ void CodeGen::genFnEpilog(BasicBlock* block)
 
         genPopCalleeSavedRegisters();
 
+#ifdef TARGET_AMD64
+        // In the case where we have an RSP frame, and no frame pointer reported in the OS unwind info,
+        // but we do have a pushed frame pointer and established frame chain, we do need to pop RBP.
+        if (doubleAlignOrFramePointerUsed())
+        {
+            inst_RV(INS_pop, REG_EBP, TYP_I_IMPL);
```

Comment:
I am not very familiar with the domain specific language in this compiler, but this is not the fix I was expecting. In the cases I analyzed, the POP RBP instruction was already there. What was missing was the UWOP_SET_FPREG unwind opcode in the prolog, when `lea rbp, [rsp + x]` is done.

Reply:
The fix requires us to go down this code path, where we didn't previously need to pop EBP, but now we do. Basically, we don't need to add cases of setting UWOP_SET_FPREG; that was already correct. We need to remove cases where we were using `lea` to pop off the frame.

Reply:
Ok, so you are following the route of not setting a frame pointer at all. And this change will cause the compiler to emit an ADD RSP instead?

Reply:
There are two frame types: RSP and RBP. For the RSP frame types, we don't emit UWOP_SET_FPREG, and restore the frame using `add rsp, constant`. However, there are still cases with RSP frames where we will save and restore the RBP register and establish a frame chain; we just don't report RBP as a frame pointer to the OS, since it isn't required for unwinding. For RBP frame types, we set UWOP_SET_FPREG and use `lea rsp, [rbp+constant]` to restore the stack. RBP frame types are typically required for alloca/stackalloc (dynamic frame size).

This change fixes the RSP frame case where we also happen to save/restore the RBP register, to not restore the stack using that RBP register.

Reply:
Yup, I understand the problem. After all it was me who reported it. I am just not seeing how this change will address the problem: making sure non-fp functions use add rsp to de-allocate the stack frame. Instead it seems as it is add a pop rbp. Am I reading it wrong?

Reply:
My read is..there was kinda a third case, where the code resembled or matched an RBP frame, but had no `UWOP_SET_FPREG` (and was correct).

Now they are following a very strict interpretation, where `lea rsp, [rbp+constant]` is only used if there is `UWOP_SET_FPREG`.

I am not sure this strictness is required but it is of course ok.